### PR TITLE
MAINT: Clarification and spelling updates

### PR DESCRIPTION
**Pull request by kaitj**
Updates the notebooks for some of simple / small fixes suggested in the reviews during the coding sprint. Mostly capitalization and rewording a sentence or two for clarification and to bring more inline across lessons.

### CODE REVIEW

File: _episodes/introduction.md [hunk 1]
```diff
@@ -125,7 +123,7 @@ tree '../data/ds000221'
 
 ## Querying a BIDS Dataset
 
-[PyBIDS](https://bids-standard.github.io/pybids/) is a Python API for
+[PyBIDS](https://bids-standard.github.io/pybids/) is a Python packageq for
```
Comment (jhlegarreta) [inline _episodes/introduction.md:126]: packageq : typo

Comment (kaitj) [reply _episodes/introduction.md]: Fixed in 0135989

File: _episodes/preprocessing.md [hunk 2]
```diff
@@ -147,9 +147,9 @@ single volume in the merged file. The first 3 values of each line refers to the
 acquisition direction, typically along the y-axis (or anterior-posterior). The
 final value is the total readout time (from center of first echo to center of
 final echo), which can be determined from values contained within the JSON
-sidecar. Each line will look similar to <code>[x y z TotalReadoutTime]</code>.
-In this case, the file, which we created, is contained within the
-<code>pedir.txt</code> file in the derivative directory.
+associated JSON metadata file. Each line will look similar to
```
Comment (jhlegarreta) [inline _episodes/preprocessing.md:150]: I would at least say "associated JSON metadata file" (named "JSON _sidecar_ file" within the BIDS specification)."

As the specification uses that term.

File: code/preprocessing.ipynb [hunk 3]
```diff
@@ -122,18 +128,13 @@
...
+    "Another file we will need to create is a text file containing the information about how the volumes were acquired. Each line in this file will pertain to a single volume in the merged file. The first 3 values of each line refers to the acquisition direction, typically along the y-axis (or anterior-posterior). The final value is the total readout time (from center of first echo to center of final echo), which can be determined from values contained within the associated JSON metadata file. Each line will look similar to `[x y z TotalReadoutTime]`. In this case, the file, which we created, is contained within the `pedir.txt` file in the derivative directory.\n",
```
Comment (jhlegarreta) [inline code/preprocessing.ipynb:137]: Same comment as in the markdown file.

File: code/preprocessing.ipynb [hunk 4]
```diff
@@ -47,34 +49,34 @@
...
+   ]
   },
   {
    "cell_type": "markdown",
+   "metadata": {},
    "source": [
-    "DIPY's `segment.mask` module will be used to create a brainmask from this. This module contains a function `median_otsu`, which can be used to segment the brain and provide a binary brainmask! Here, a brainmask will be created using the first non-diffusion volume of the data. We will save this brainmask to be used in our later future preprocessing steps. After creating the brainmask, we will start to correct for distortions in our images."
-   ],
-   "metadata": {}
+    "DIPY's `segment.mask` module will be used to create a brainmask from this. This module contains a function `median_otsu`, which can be used to segment the brain and provide a binary brainmask! Here, a brainmask will be created using the first B0 volume of the data. We will save this brainmask to be used in our later future preprocessing steps. After creating the brainmask, we will start to correct for distortions in our images."
```
Comment (jhlegarreta) [inline code/preprocessing.ipynb:72]: If we want to shorten the "non-diffusion volume" with "B0 volume", it should already have been introduced, e.g. here:
https://github.com/carpentries-incubator/SDC-BIDS-dMRI/blob/main/_episodes/introduction.md?plain=1#L273

(e.g. "(...) commonly referred to as B0 volume or image").

Also, note that in the very preprocessing lesson, it is written in lowercases, e.g.
https://github.com/carpentries-incubator/SDC-BIDS-dMRI/blob/main/_episodes/introduction.md?plain=1#L273
or
https://github.com/carpentries-incubator/SDC-BIDS-dMRI/blob/main/_episodes/introduction.md?plain=1#L273

for example. I have not checked other lessons. All appearances should be written with the same case.

Both in the markdown and the Jupyter notebook code and comments.

File: code/introduction.ipynb [hunk 5]
```diff
@@ -123,36 +127,33 @@
...
    "metadata": {
     "jupyter": {
      "outputs_hidden": true
     },
     "scrolled": true
-   }
+   },
+   "outputs": [],
+   "source": [
+    "dwi = layout.get(subject='010006', suffix='dwi', extension='nii.gz', return_type='file')[0]\n",
```
Comment (jhlegarreta) [inline code/introduction.ipynb:150]: Leading dot extension missing. Probably rebasing on `main` would solve this as GitHub says that files are conflicting with `main`.

File: _episodes/preprocessing.md [hunk 6]
```diff
@@ -38,7 +38,7 @@ dMRI has some similar challenges to fMRI preprocessing, as well as some unique
 [ones](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3366862/).
 
 Our preprocesssing of this data will consist of following steps and will make
-use of sub-010006:
+use of `sub-010006`:
```
Comment (jhlegarreta) [inline _episodes/preprocessing.md:41]: If this is done in the Jupyter notebook, it should be done in the markdown file as well.

File: code/preprocessing.ipynb [hunk 7]
```diff
@@ -12,26 +13,27 @@
     "\n",
     "dMRI has some similar challenges to fMRI preprocessing, as well as some unique [ones](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3366862/).\n",
     "\n",
-    "Our preprocesssing of this data will consist of following steps and will make use of sub-010006:\n",
+    "Our preprocesssing of this data will consist of following steps and will make use of `sub-010006`:\n",
```
Comment (jhlegarreta) [inline code/preprocessing.ipynb:16]: If this is done in the Jupyter notebook, it should be done in the markdown file as well.

File: _episodes/preprocessing.md (hunk 6 above)
Comment (jhlegarreta) [reply _episodes/preprocessing.md:41]: Also, the sentence does not make much sense/should be reworded, e.g.
"Our preprocesssing of this data will consist of following steps:
(...)

The same subject (`sub-010006`) will be used throughout the whole episode."
(or "We will be using the same subject (`sub-010006`) throughout the whole episode.")

File: code/introduction.ipynb [hunk 8]
```diff
@@ -255,104 +257,102 @@
...
-    "dwi_data = layout.get(suffix='dwi', extension='nii.gz', return_type='file')"
-   ],
-   "outputs": [],
    "metadata": {
     "solution2": "hidden"
-   }
+   },
+   "outputs": [],
+   "source": [
+    "dwi_data = layout.get(suffix='dwi', extension='nii.gz', return_type='file')"
```
Comment (jhlegarreta) [inline code/introduction.ipynb:343]: Leading dot extension missing. Probably rebasing on `main` would solve this as GitHub says that files are conflicting with `main`.

File: code/preprocessing.ipynb [hunk 9]
```diff
@@ -193,20 +195,19 @@
...
     "\n",
-    "Here, we will demonstrate the application of `eddy` following the `topup` correction step, by making use of both the uncorrected diffusion data, as well as distortion corrections from the previous step. Additionally, a text file, which maps each of the volumes to one of the corresponding acquisition directions from the `pedir.txt` file will have to be created. Finally, similar to `topup`, there are also a number of input parameters which have to be specified:\n",
+    "Here, we will demonstrate the application of `eddy` following the `topup` correction step, by making use of both the uncorrected diffusion data, as well as estimated warpfield from `topup`. Additionally, a text file, which maps each of the volumes to one of the corresponding acquisition directions from the `pedir.txt` file will have to be created. Finally, similar to `topup`, there are also a number of input parameters which have to be specified:\n",
```
Comment (jhlegarreta) [inline code/preprocessing.ipynb:210]: Should be done in the markdown file as well.

File: code/preprocessing.ipynb [hunk 10]
```diff
@@ -245,24 +247,28 @@
...
     "\n",
-    "First, we will create a brainmask of the anatomical image using the second inversion. To do this, we will use `FSL` `bet` twice. The first call to `bet` will create a general skullstripped brain. Upon inspection, we can note that there is still some residual areas of the image which were included in the first pass. Calling `bet` a second time, we get a better outline of the brain and brainmask, which we can use for further processing."
-   ],
-   "metadata": {}
+    "First, we will create a brainmask from the second inversion of the anatomical acquisition (e.g. T1-weighted). To do this, we will use `FSL` `bet` twice. The first call to `bet` will create a general skullstripped brain. Upon inspection, we can note that there is still some residual areas of the image which were included in the first pass. Calling `bet` a second time, we get a better outline of the brain and brainmask, which we can use for further processing."
```
Comment (jhlegarreta) [inline code/preprocessing.ipynb:260]: Should be done in the markdown file as well.

File: code/preprocessing.ipynb [hunk 11]
```diff
@@ -145,16 +146,17 @@
     "```\n",
     "\n",
     "With these two inputs, the next step is to make the call to `topup` to estimate the susceptibility-induced field. Within the call, a few parameters are used. Briefly:\n",
-    " * `--imain` specifies the previously merged volume\n",
+    " * `--imain` specifies the previously merged volume.\n",
```
Comment (jhlegarreta) [inline code/preprocessing.ipynb:149]: Should be done in the markdown file as well.

And if this is the preferred style, should be done at once in all episodes and notebooks.

File: code/preprocessing.ipynb [hunk 12]
```diff
@@ -145,16 +146,17 @@
...
     " * `--datain` specifies the text file containing the information regarding the acquisition.\n",
     " * `--config=b02b0.cnf` makes use of a predefined config file supplied with `topup`, which contains parameters useful to registering with good $b = 0 s/mm^2$ images.\n",
-    " * `--out` defines the output files containing the spline coefficients for the induced field, as well as subject movement parameters"
-   ],
-   "metadata": {}
+    " * `--out` defines the output files containing the spline coefficients for the induced field, as well as subject movement parameters."
```
Comment (jhlegarreta) [inline code/preprocessing.ipynb:152]: Same comment as above.

File: code/introduction.ipynb [hunk 13]
```diff
@@ -2,40 +2,41 @@
...
+    "Diffusion MRI is a popular technique to study the brain's white matter. To do so, sequences sensitive to random, microscropic motion of water protons are applied. The diffusion within biological structures, such as the brain, are often restricted due to barriers (e.g. cell membranes), resulting in a preferred direction of diffusion (anisotropy). A typical diffusion MRI scan will acquire multiple volumes that are sensitive to a particular diffusion direction and result in diffusion-weighted images (DWI). Diffusion that exhibits directionality in the same direction result in an attenuated signal. With further processing (to be discussed later in the lesson), the acquired images can provide measurements which are related to the microscopic changes and estimate white matter trajectories. Images with no diffusion weighting are also acquired as part of the acquisition protocol.\n",
     "\n",
     "![fiber_configurations](../fig/introduction/DiffusionDirections.png) \\\n",
     "Diffusion along X, Y, and Z directions"
-   ],
-   "metadata": {}
```
Comment (jhlegarreta) [inline code/introduction.ipynb:17]: Adding all these style changes makes reviewing difficult. I guess these are due to different editor/Jupyter versions being used, and we are likely to go around in circles for every PR. I grant excluding these makes it longer to git add changes, but I think it is better. If they need to be included, I'd add them in a separate, style-only commit.

File: code/introduction.ipynb [hunk 14]
```diff
@@ -377,4 +377,4 @@
  },
  "nbformat": 4,
  "nbformat_minor": 4
-}
```
Comment (jhlegarreta) [inline code/introduction.ipynb:380]: Same comment about the style for the newline at the end of the file.

File: code/introduction.ipynb (hunk 13 above)
Comment (jhlegarreta) [reply code/introduction.ipynb:17]: Seems like these are already present in the Markdown file, and this is trying to sync the Jupyter Notebook with them. Also for the sake of simplicity and going forward, keeping them in this commit.

File: _episodes/preprocessing.md (hunk 6 above)
Comment (jhlegarreta) [reply _episodes/preprocessing.md]: Done in 7b3a216.

File: _episodes/preprocessing.md (hunk 2 above)
Comment (jhlegarreta) [reply _episodes/preprocessing.md]: Done in 7b3a216.

File: code/preprocessing.ipynb (hunk 7 above)
Comment (jhlegarreta) [reply code/preprocessing.ipynb]: Reworded on both the Jupyter Notebook and Markdown in 7b3a216.

File: code/preprocessing.ipynb (hunk 3 above)
Comment (jhlegarreta) [reply code/preprocessing.ipynb]: Done in 7b3a216.

File: code/preprocessing.ipynb (hunk 11 above)
Comment (jhlegarreta) [reply code/preprocessing.ipynb:152]: Done in 7b3a216. Doing this across all episodes/notebooks is left for a future PR in order to go forward.

File: code/preprocessing.ipynb (hunk 12 above)
Comment (jhlegarreta) [reply code/preprocessing.ipynb:155]: Done in 7b3a216.

File: code/preprocessing.ipynb (hunk 9 above)
Comment (jhlegarreta) [reply code/preprocessing.ipynb:213]: Done in 7b3a216.

File: code/preprocessing.ipynb (hunk 10 above)
Comment (jhlegarreta) [reply code/preprocessing.ipynb:263]: Done in 7b3a216.

File: code/introduction.ipynb (hunk 14 above)
Comment (jhlegarreta) [reply code/introduction.ipynb:377]: Added back in de4d3a4.

File: code/introduction.ipynb (hunk 8 above)
Comment (jhlegarreta) [reply code/introduction.ipynb]: Added back in de4d3a4.

File: code/introduction.ipynb (hunk 5 above)
Comment (jhlegarreta) [reply code/introduction.ipynb]: Added back in de4d3a4.

File: code/preprocessing.ipynb (hunk 4 above)
Comment (jhlegarreta) [reply code/preprocessing.ipynb:72]: Done in d38d768. Left as `b0` in the code comments, as it is typically used in lowercase in the variable names.

**Comment by jhlegarreta** [comment]
Test errors stem from the fact that we are using DIPY 1.3.0, and NumPY 1.24, and the latter removed all support for `np.float`.
https://github.com/carpentries-incubator/SDC-BIDS-dMRI/actions/runs/3773484312/jobs/6414959616#step:13:102

https://numpy.org/doc/stable/release/1.24.0-notes.html#expired-deprecations

`gtab` not existing is a consequence of the above:
https://github.com/carpentries-incubator/SDC-BIDS-dMRI/actions/runs/3773484312/jobs/6414959616#step:13:134

DIPY 1.4.1 contains the necessary fixes. Upgrading DIPY in a separate PR, then will rebase this PR.

**Comment by jhlegarreta** [comment]
Cross-referencing PR #218.

**Comment by jhlegarreta** [comment]
Python 3.9 exceeded the maximum execution time of 6 hours; Python 3.7 and 3.8 succeeded:
https://github.com/carpentries-incubator/SDC-BIDS-dMRI/actions/runs/3778995767

Merging.

**Pull request merged**